AEPCore/Tests/FunctionalTests/RulesEngineHistoricalTests.swift

```diff
@@ -0,0 +1,679 @@
+        //          "schema": "https://ns.adobe.com/personalization/eventHistoryOperation",
+        //          "data": {
+        //              "operation": "invalid",
+        //              "content": {
+        //                  "key1": "value1"
+        //              }
+        //          }
+        //        }
+        //    --------------------------------------
+        resetRulesEngine(withNewRules: "consequence_rules_testDataOperationInvalid")
+
+        // When:
+        rulesEngine.process(event: defaultEvent)
+
+        // Then: Event history consequence event:
+        // Should not attempt to record into event history
+        XCTAssertFalse(mockRuntime.recordHistoricalEventCalled)
+        // Should not be dispatched
+        XCTAssertEqual(0, mockRuntime.dispatchedEvents.count)
+    }
+
+    func testSchemaConsequenceInsert_doesNotDispatchOrRecord_whenContentIsMissing() {
+        // Given: a schema type rule consequence with an invalid format for event history operation: ('content' missing)
+        //    ---------- schema based consequence details ----------
+        //        "detail": {
+        //          "id": "test-id",
+        //          "schema": "https://ns.adobe.com/personalization/eventHistoryOperation",
+        //          "data": {
+        //              "operation": "insert"
+        //          }
+        //        }
+        //    --------------------------------------
+        resetRulesEngine(withNewRules: "consequence_rules_testDataContentMissing")
+
+        // When:
+        rulesEngine.process(event: defaultEvent)
+
+        // Then: Event history consequence event:
+        // Should not attempt to record into event history
+        XCTAssertFalse(mockRuntime.recordHistoricalEventCalled)
+        // Should not be dispatched
+        XCTAssertEqual(0, mockRuntime.dispatchedEvents.count)
+    }
+
+    func testSchemaConsequenceInsert_doesNotDispatchOrRecord_whenContentIsNull() {
+        // Given: a schema type rule consequence with an invalid format for event history operation: ('content' null)
+        //    ---------- schema based consequence details ----------
+        //        "detail": {
+        //          "id": "test-id",
+        //          "schema": "https://ns.adobe.com/personalization/eventHistoryOperation",
+        //          "data": {
+        //              "operation": "insert",
+        //              "content": null
+        //          }
+        //        }
+        //    --------------------------------------
+        resetRulesEngine(withNewRules: "consequence_rules_testDataContentNull")
+
+        // When:
+        rulesEngine.process(event: defaultEvent)
+
+        // Then: Event history consequence event:
+        // Should not attempt to record into event history
+        XCTAssertFalse(mockRuntime.recordHistoricalEventCalled)
+        // Should not be dispatched
+        XCTAssertEqual(0, mockRuntime.dispatchedEvents.count)
+    }
+}
```

Comment:
nit : Can we also add a test case for `testSchemaConsequenceInsert_doesNotDispatchOrRecord_whenContentIsEmpty`

Reply:
Ah I missed it. Should be fine where it is now too